apps/meteor/app/api/server/v1/groups.ts

```diff
@@ -389,21 +389,28 @@ API.v1.addRoute(
 
 API.v1.addRoute(
 	'groups.files',
-	{ authRequired: true },
+	{ authRequired: true, validateParams: isGroupsFilesProps },
 	{
 		async get() {
+			const { typeGroup, name, roomId, roomName } = this.queryParams;
+
 			const findResult = await findPrivateGroupByIdOrName({
-				params: this.queryParams,
+				params: roomId ? { roomId } : { roomName },
 				userId: this.userId,
 				checkedArchived: false,
 			});
 
 			const { offset, count } = await getPaginationItems(this.queryParams);
 			const { sort, fields, query } = await this.parseJsonQuery();
 
-			const ourQuery = Object.assign({}, query, { rid: findResult.rid });
+			const filter = {
+				rid: findResult.rid,
+				...query,
```

Comment:
The query object is spread after setting rid. If the query also has an rid property, it will override the previously set rid. Hence, I am concerned it can be used to fetch files for arbitrary rooms irrespective of the passed roomId query param.

Reply:
I just confused myself a bit. Just moved the spread, thanks for the review man!